src/visip/action/GMSH_action.py

```diff
@@ -0,0 +1,185 @@
+    return my_Mesh
+
+
+@action_def
+def ExtractRegionElements(msh: MeshGMSH, regions: List[str]) -> SubMesh:
+    """
+    Extracts chosen regions from mesh and creates SubMesh from those regions.
+    :param msh: whole mesh
+    :param regions: regions to extract
+    :return: extracted regions as SubMesh  - List[Tuple[int, Element]]
+    """
+    region_id_dim = []
+    for region in regions:
+        if region in msh.regions.keys():
+            region_val = [val for name, val in msh.regions.items() if name == region]
+            region_id_dim.append(*region_val)
+
+    element_id = []
+    elements = []
+    region_dim_only = [x[0] for x in region_id_dim]
+    for ele_id, ele_val in msh.elements.items():
+        for region_dim in region_dim_only:
+            if ele_val.region_id == region_dim:
+                element_id.append(ele_id)
+                elements.append(ele_val)
+
+    result = list(zip(element_id, elements))
+    return result
+
+
+@action_def
+def ExtractRegionIds(submsh: SubMesh) -> List[int]:
+    RegionIds = []
+    for Element_id, Element in submsh:
+        RegionIds.append(Element.region_id)
+    return RegionIds
+
+
+@action_def
+def EleIds(msh: SubMesh) -> List[int]:
+    return [x[0] for x in msh]
+
+
+@action_def
+def Barycenter(mesh: MeshGMSH, msh: SubMesh) -> np.ndarray:  # output?
+    pp_bary = []
+    for id, elem in msh:
+        keys = ['x', 'y', 'z']
+        pom_dict = {key: [] for key in keys}
+        for point_id in elem.nodes:
+            point = mesh.nodes[point_id]
+            pom_dict['x'].append(point.x)
+            pom_dict['y'].append(point.y)
+            pom_dict['z'].append(point.z)
+        bary_x = np.average(pom_dict['x'])
+        bary_y = np.average(pom_dict['y'])
+        bary_z = np.average(pom_dict['z'])
+
+        pp_bary.append([bary_x, bary_y, bary_z])
+
+    barycenter = np.array(pp_bary)
+    return barycenter
+
+
+# @action_def
+# def sampler_fn(seed: int) -> Any:
+#     return np.random.seed(seed)
+
+
+@action_def
+def field_mesh_sampler(all_fields: cf.Fields, outer_field_names: List[str],
+                       barycenters: np.ndarray, region_ids: List[int]) -> Any:  # should return @action
+
+    # fields = cf.Fields(all_fields)
+    all_fields.set_outer_fields(outer_field_names)
+    all_fields.set_points(barycenters, region_ids=region_ids)
+
+    def sampler_fn(seed: int) -> Any:
+        np.random.seed(seed)
+        return all_fields.sample()
+
+    return wf.action_def(sampler_fn)
+
+
+@action_def
+def make_fields() -> cf.Fields:
+    return cf.Fields([
+        cf.Field('conductivity', cf.FourierSpatialCorrelatedField('gauss', dim=3, corr_length=0.125, log=True)),
+    ]) # Field s dim=3 není implementován. Moje chyba?
+    # return cf.Fields([
+    #     cf.Field(name='conductivity', field=1),
+    #     cf.Field(name='conductivity1', field=2)
+    # ])
+
+
+@action_def
+def write_fields(mesh: MeshGMSH, ele_ids: List[int], sampler: wf.Any) -> Any:
```

Comment:
`sampler` parameter should be named `sample`, it is a value returned by the field sampler